internal/charmstore/search.go

```diff
@@ -230,6 +236,21 @@ func (si *SearchIndex) update(doc *SearchDoc) error {
 	if err != nil && err != elasticsearch.ErrConflict {
 		return errgo.Mask(err)
 	}
+	if doc.Entity.URL.Series != "" {
+		return nil
+	}
+	// This document represents a multi-series charm it might be
+	// replacing existing documents for previous single-series
+	// charms. Remove any documents that this replaces.
+	for _, series := range doc.Entity.SupportedSeries {
```

Comment:
Is this actually sufficient? What if there's an old "precise" charm but the multi-series charm that's been uploaded doesn't support precise?